src/stdune/path.ml

```diff
@@ -866,6 +866,7 @@ let readdir_unsorted t = Sys.readdir (to_string t) |> Array.to_list
 let is_directory t =
   try Sys.is_directory (to_string t)
   with Sys_error _ -> false
+let is_file t = not (is_directory t)
```

Comment:
does it work for things like named pipes and sockets?

Reply:
Probably not, but this is why I'm introducing this function here. So that we can improve it to take care of these corner cases.